### PR TITLE
Added demos: Histogram Equalization and RGB to Grayscale conversion

### PR DESCRIPTION
**Pull request by mgautam98**
Added the following demos
1. histogram (equalization, adaptive equalization, gamma correction)
2. RGB to grayscale conversion

Since build was failing I couldn't test it.
#47 #48  Are not needed anymore

### CODE REVIEW

**Comment by johnnychen94** [comment]
The CI looks weird, will check that later today.

**Comment by johnnychen94** [comment]
It was a incompatible change introduced by Literate v1.5, it should be fixed in DemoCards v0.2.2 now.

**Comment by johnnychen94** [comment]
@mgautam98 you can now build the docs locally. (Don't forget a `]up` to update your local `Manifest.toml`)

**Comment by mgautam98** [comment]
@johnnychen94 I tested it on my local machine and fixed some error

File: docs/examples/color_channels/rgb_grayscale.jl [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+using Images, TestImages
+
+rgb_image = testimage("lighthouse")
+
+# I = Gray.(RGB) converts an RGB image to Grayscale. 
+
+gray_image = Gray.(rgb_image)
+hcat(rgb_image, gray_image)
```
Comment (johnnychen94) [inline docs/examples/color_channels/rgb_grayscale.jl:16]:
```diff
- hcat(rgb_image, gray_image)
+ mosaicview(rgb_image, gray_image; nrow = 1)
```

I hope we can advertise the new util `mosaicview` here, which is reexported by `ImageCore`.

https://github.com/JuliaArrays/MosaicViews.jl#compare-two-images

File: docs/examples/spatial_transformation/histogram_equalization.jl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+img = testimage("moonsurface")
+
+# Now we will apply Histogram equalisation, gamma correction and Adaptive histogram equalisation 
+# method to enhance contrast of the image
+
+hist_equal = adjust_histogram(img, Equalization(nbins = 256))
+gamma_correction = adjust_histogram(img, GammaCorrection(gamma = 2))
+hist_adapt = adjust_histogram(img, AdaptiveEqualization(nbins = 256, rblocks = 4, cblocks = 4, clip = 0.2))
+
+hcat(img, hist_equal, gamma_correction, hist_adapt)
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/histogram_equalization.jl:27]:
```diff
- hcat(img, hist_equal, gamma_correction, hist_adapt)
+ mosaicview(img, hist_equal, gamma_correction, hist_adapt; nrow = 1)
```

File: docs/examples/spatial_transformation/histogram_equalization.jl [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+# --- save covers --- #src
+using Images #src
```
Comment (johnnychen94) [inline docs/examples/spatial_transformation/histogram_equalization.jl:30]:
```suggestion
using FileIO #src
```

**Comment by mgautam98** [comment]
Thanks for the suggestions I have made the changes

File: docs/examples/color_channels/rgb_grayscale.jl [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# ---
+
+# This example illustrates RGB to Grayscale Conversion
+
+
+using Images, TestImages
+
+rgb_image = testimage("lighthouse")
+
+# I = Gray.(rgb_image) converts an RGB image to Grayscale.
```
Comment (johnnychen94) [inline docs/examples/color_channels/rgb_grayscale.jl:13]:
```suggestion
# `I = Gray.(rgb_image)` converts an RGB image to Grayscale. 
```

😄

File: docs/examples/color_channels/rgb_grayscale.jl [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# ---
+# title: RGB to GrayScale
+# cover: assets/rgb_grayscale.png
+# ---
+
+# This example illustrates RGB to Grayscale Conversion
+
+
+using Images, TestImages
```
Comment (johnnychen94) [inline docs/examples/color_channels/rgb_grayscale.jl:9]:
```suggestion
using ImageCore, TestImages
```

Sorry that I missed this.

**Comment by johnnychen94** [comment]
Thanks @mgautam98 !

**Pull request merged**